bucket-policy-parser.go

```diff
@@ -264,6 +270,28 @@ func checkBucketPolicyResources(bucket string, bucketPolicy BucketPolicy) APIErr
 	return ErrNone
 }
 
+
+// removeNonePseudoStatements - removes the pseudo 'none' statements (s3:None and s3:NoneBucket) from bucket policy
+func removePseudoNoneStatements(bucketPolicy BucketPolicy) (statements []policyStatement, changed bool) {
+	// Swap all of the none statements to the beginning and return slice of the remaining portion
+	swapIndex := 0
+	for index, statement := range bucketPolicy.Statements {
+		for _, action := range statement.Actions {
+			if strings.HasPrefix(action, "s3:None") {
```

Comment:
> I labelled them as "Pseudo statements". I'm not sure if this is the best way to solve the bug but the only other option is checking for nesting on the client side and that may be redundant since the server is already doing it. So I tried fitting 'none' into the server's nesting check by making a statement for it.
> These statements are deleted directly after checking, they are never saved.

This isn't the best way to do it, we are introducing new fields which are not necessary and we are also making some custom policy actions.